### PR TITLE
Fix potential GC close issue

### PR DESCRIPTION
**Pull request by cclerget**
**Description of the Pull Request (PR):**

As title say

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](../CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](../CHANGELOG.md) if necessary according to the [Contribution Guidelines](../CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](../CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](../CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

File: internal/pkg/util/exec/pipe_linux.go [hunk 1]
```diff
@@ -62,26 +61,26 @@ func setPipe(data []byte) (*os.File, error) {
...
-		return nil, fmt.Errorf("failed to duplicate pipe file descriptor: %s", err)
+		return -1, fmt.Errorf("failed to duplicate pipe file descriptor: %s", err)
 	}
 
 	if n, err := syscall.Write(fd[0], data); err != nil || n != len(data) {
-		return nil, fmt.Errorf("failed to write data to stdin: %s", err)
+		return -1, fmt.Errorf("failed to write data to stdin: %s", err)
 	}
 
-	return os.NewFile(uintptr(pipeFd), "pipefd"), err
```
Comment (sashayakovtseva) [inline internal/pkg/util/exec/pipe_linux.go:77]: So without associated `*os.File` fd is not closed by the runtime?

File: internal/pkg/runtime/engines/oci/prepare.go [hunk 2]
```diff
@@ -171,21 +180,16 @@ func (e *EngineOperations) PrepareConfig(starterConfig *starter.Config) error {
 			e.EngineConfig.MasterPts = int(master.Fd())
 			e.EngineConfig.SlavePts = int(slave.Fd())
 		} else {
-			r, w, err := os.Pipe()
-			if err != nil {
+			flags := syscall.O_CLOEXEC
+			if err := syscall.Pipe2(e.EngineConfig.OutputStreams[0:], flags); err != nil {
```
Comment (sashayakovtseva) [inline internal/pkg/runtime/engines/oci/prepare.go:184]: Can be simplified to `e.EngineConfig.OutputStreams[:]` to be more go-way

File: internal/pkg/util/exec/pipe_linux.go (hunk 1 above)
Comment (cclerget) [reply internal/pkg/util/exec/pipe_linux.go:77]: Yes

File: internal/pkg/runtime/engines/oci/prepare.go (hunk 2 above)
Comment (cclerget) [reply internal/pkg/runtime/engines/oci/prepare.go:184]: Copied from https://golang.org/src/os/pipe_linux.go?s=319:360#L1 :)

**Review by sashayakovtseva — APPROVED** [review]

**Pull request merged**